### PR TITLE
Replace react/lib/* deps missing in React 0.14

### PR DESCRIPTION
**Pull request by jeremiahrhall**
Replaced:

- `react/lib/ExecutionEnvironment` with `exenv`
- `react/lib/invariant` with `invariant`
- `react/lib/warning` with `warning`

This gets us online using this branch in React 0.14.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 v0.13.3 - Wed, 29 Apr 2015 01:47:08 GMT
```
Comment (jcasner) [inline CHANGELOG.md]: Looks like we never got comments for `v0.13.4`; let's not make that mistake with `v0.13.5`

**Comment by jcasner** [comment]
Might be a red herring, but I'm getting:
```
Module parse failed: /Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/@opengov/react-router/modules/components/RouteHandler.js Unexpected token (58:21)
You may need an appropriate loader to handle this file type.
SyntaxError: Unexpected token (58:21)
    at Parser.pp.raise (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:920:13)
    at Parser.pp.unexpected (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:1483:8)
    at Parser.pp.parseExprAtom (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:330:12)
    at Parser.pp.parseExprSubscripts (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:225:19)
    at Parser.pp.parseMaybeUnary (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:204:17)
    at Parser.pp.parseExprOps (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:151:19)
    at Parser.pp.parseMaybeConditional (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:133:19)
    at Parser.pp.parseMaybeAssign (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:110:19)
    at Parser.pp.parseMaybeConditional (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:138:28)
    at Parser.pp.parseMaybeAssign (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:110:19)
    at Parser.pp.parseExpression (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:86:19)
    at Parser.pp.parseReturnStatement (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:1854:26)
    at Parser.pp.parseStatement (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:1719:19)
    at Parser.pp.parseBlock (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:1991:21)
    at Parser.pp.parseFunctionBody (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:607:22)
    at Parser.pp.parseMethod (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:576:8)
    at Parser.pp.parseClassMethod (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:2137:23)
    at Parser.pp.parseClass (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:2122:10)
    at Parser.pp.parseStatement (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:1715:19)
    at Parser.pp.parseTopLevel (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:1648:21)
    at Parser.parse (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:1616:17)
    at Object.parse (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/node_modules/acorn/dist/acorn.js:882:44)
    at Parser.parse (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/lib/Parser.js:902:15)
    at DependenciesBlock.<anonymous> (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack/lib/NormalModule.js:104:16)
    at DependenciesBlock.onModuleBuild (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack-core/lib/NormalModuleMixin.js:310:10)
    at nextLoader (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack-core/lib/NormalModuleMixin.js:275:25)
    at /Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/webpack-core/lib/NormalModuleMixin.js:259:5
    at Storage.finished (/Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:38:16)
    at /Users/jaredcasner/Code/OpenGov/ReportApp/node_modules/graceful-fs/graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:380:3)
 @ ./~/@opengov/react-router/modules/index.js 6:24-60```
when I launch ReportApp locally pointing at this commit. I then get stuck in an infinite redirect loop

**Comment by frodosamoa** [comment]
How did you do the replacing? Was it all manual?

**Comment by jeremiahrhall** [comment]
Webpack was throwing compile errors until these particular items were replaced. It's very greedy about finding requires even if the codepath isn't executed. Replacement was done specifically with find/replace in atom in the `modules` directory. The rest of the JS changes are the result of the build script.

**Comment by jeremiahrhall** [comment]
I manually verified that the remaining `react/lib` dependency `react/lib/Object.assign` is still present in React 0.14.

**Pull request closed without merging**